### PR TITLE
feat: Using ###CHR### as pattern to get multiple custom files

### PR DESCRIPTION
**Pull request by diegomscoelho**
Ticket: [JIRA Card](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4753)
## Description

To facilitate user usage of `--custom` flag, we are introducing `###CHR###` pattern to automatically loop through known chromosomes files.

## Technical description

1) `get_all_custom` was updated to get all chromosomes combinations when user states `###CHR###` in a filename.
2) If no chromosome were found using this pattern, `--custom` usage will die() stating no file were found to this pattern.
3) Valid chromosomes were limited to 1 to 22 + X, Y and MT.

## Test

1) Try using,new pattern, i.e:
```sh
--custom /<PATH-TO>/gnomad.genomes.v3.1.2.sites.chr###CHR###_trimmed_info.vcf.bgz,gnomADg,vcf,exact,,AC,AN,AF,AF_popmax`
```
2) Also try to give wrong filename to pattern, i.e:
```sh
# An error should pop up.
/<PATH-TO>/wrong_name.chr###CHR###.vcf.bgz,gnomADg,vcf,exact,,AC,AN,AF,AF_popmax
```
3) Also try two custom flags, one with new pattern and one without to see if it works.

P.S.: This PR will be nicely fit to https://github.com/Ensembl/ensembl-vep/pull/1159

### CODE REVIEW

File: modules/Bio/EnsEMBL/VEP/AnnotationSourceAdaptor.pm [hunk 1]
```diff
@@ -224,11 +225,27 @@ sub get_all_custom {
...
+        my $new_file = $file;
+        my $new_opts = { %$opts };
+        $new_file =~ s/\#\#\#CHR\#\#\#/$chr/;
+        next unless -e $new_file;
+        $new_opts->{file} = $new_file;
+        push @as, Bio::EnsEMBL::VEP::AnnotationSource::File->new($new_opts);
+      }
+
+      # Non-match ###CHR### pattern scenario
+      die "Error: No file with pattern $file were found" unless @as;
```
Comment (nuno-agostinho) [inline modules/Bio/EnsEMBL/VEP/AnnotationSourceAdaptor.pm:241]:
```suggestion
      die "Error: No files with pattern $file were found\n" unless @as;
```

**Review by nuno-agostinho — COMMENTED** [review]
Thanks for these changes, @diegomscoelho! I have some suggestions below, please tell me if you want to discuss something in more detail.

| Condition | Expected | Note | Check |
| - | - | - | - |
| Not using pattern (custom file for chr 2) | Output | Custom annotation for variants in chr2 | 🟢 |
| Using pattern (custom files for chr 1-22, X and Y) | Output | Custom annotation for all variants | 🟢 |
| Using pattern (custom files for chr 1 and 2) | Output | Custom annotation for variants in chr 1-2 | 🟢 |
| Custom FTP file without pattern ([docs](https://www.ensembl.org/info/docs/tools/vep/script/vep_custom.html); [example files](http://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh38/variation_genotype/ggvp/)) | Output | Custom annotation | 🟢 |
| Using pattern with FTP files ([docs](https://www.ensembl.org/info/docs/tools/vep/script/vep_custom.html); [example files](http://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh38/variation_genotype/ggvp/)) | Output | **Currently gives an error, should this be supported?** | 🔴 |
| Multiple `--custom` flags (some with patterns, others without) | Output | Custom annotation from multiple files | 🟢 |
| Using pattern with wrong file name | Error | `Error: No file with pattern test_chr###CHR###.vcf.gz were found` | 🟢 |

File: .travis.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ addons:
       - mysql-client-core-5.6
 
 before_install:
-  - export ENSEMBL_BRANCH=master
```
[the remark this answers is not in the document]
Comment (diegomscoelho) [reply .travis.yml]: Reverted!

**Comment by diegomscoelho** [comment]
Fixed: Using pattern with FTP files case. Thanks for the help @nuno-agostinho .

File: modules/Bio/EnsEMBL/VEP/AnnotationSourceAdaptor.pm [hunk 3]
```diff
@@ -70,6 +70,7 @@ use Bio::EnsEMBL::VEP::AnnotationSource::Database::Variation;
 use Bio::EnsEMBL::VEP::AnnotationSource::Database::StructuralVariation;
 use Bio::EnsEMBL::VEP::AnnotationSource::File;
 
+my @VALID_CHRS = ((1..22), qw(X Y MT));
```
[the remark this answers is not in the document]
Comment (nuno-agostinho) [reply modules/Bio/EnsEMBL/VEP/AnnotationSourceAdaptor.pm:75]: We can retrieve a list of species-based chromosomes using `keys $self->chr_lengths()` (not sure if this is the best approach, given that this also returns non-standard chromosomes, e.g., `KI270590.1` and `CHR_HSCHR7_2_CTG6`).

**Pull request merged**